verifier/testreport.py

```diff
@@ -332,6 +341,58 @@ def create_html_report(self):
     file.close()
     return html_output
 
+  def setsize(s):
+    return len(s)
+
+  def characterizeFailuresByOptions(self):
+    # User self.failing_tests, looking at options
+    results = {}
+    fail_combos = {}
+    for test in self.failing_tests:
+      # Get input_data, if available
+      label = test['label']
+      if test.get('input_data'):
+        # Look at locale
+        input_data = test.get('input_data')
+
+        if input_data.get('locale'):
+          failure_combo = 'locale' + ':' + input_data.get('locale')
+          if failure_combo not in results:
+            results[failure_combo] = set()
```

Comment:
Consider [`defaultdict`](https://docs.python.org/3/library/collections.html#collections.defaultdict) for this type of thing.